### PR TITLE
fix(medeusa): Transform query includes options should only be added to allowed props if there is already at least one allowed props

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
when `fields` only contain includes options, it should return the entire object plus the include options. If the fields contains the included options + other fields, it should only return the requested fields + the included options

### CODE REVIEW

**Comment by olivermrbl** [comment]
@adrien2p Please move base to `master` so we can cut a patch release with this :)

**Comment by adrien2p** [comment]
@olivermrbl moved to master 💪

**Comment by adrien2p** [comment]
LOL I ve lost a commit

**Comment by adrien2p** [comment]
@olivermrbl just thought, on master it is working, it was only on develop that it was not 😂 this is why I ve lost a commit

**Comment by adrien2p** [comment]
@kasperkristensen let me know if everything is ok or if something is wrong

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Comment by kasperkristensen** [comment]
@adrien2p I am still not getting the full order back, if there is also an `expand` on the query, which breaks admin as we are some places doing queries like this `{ fields: "returnable_items", expand: "..." }`

**Comment by kasperkristensen** [comment]
Will this PR still be merged into develop, and not master? If that's the case I would like to push my changes to admin to this PR as well as it makes it easier to test out that everything works as expected. @olivermrbl @adrien2p

**Comment by adrien2p** [comment]
Yes it will be pused to develop as the changes does not exists on master in regards to the middleware, I am fine with having the admin here as well. Will you cherry pick the commits if we go that way?

**Review by kasperkristensen — APPROVED** [review]
LGTM!

I've opened a PR based on this for the admin #3383, currently, it is opened against this branch, but depending on whether you would prefer to merge this as is I can point it towards `develop` after this is merged

**Comment by adrien2p** [comment]
@olivermrbl should we merge this one so that kasper can continue?

File: integration-tests/api/__tests__/admin/order/order.js
Comment (olivermrbl) [inline integration-tests/api/__tests__/admin/order/order.js:2485]: question: Could we match the exact keys instead to make it even more robust?

**Review by olivermrbl — APPROVED** [review]
LGTM!

File: integration-tests/api/__tests__/admin/order/order.js
Comment (adrien2p) [reply integration-tests/api/__tests__/admin/order/order.js:2485]: Yes sure 😊

**Comment by olivermrbl** [comment]
Tests seem to be failing :)

**Comment by adrien2p** [comment]
@olivermrbl fixed, just waiting for the CI

**Pull request merged**
